kore/src/Kore/Step/Simplification/Forall.hs

```diff
@@ -98,8 +118,25 @@ makeEvaluate
 makeEvaluate variable patt
   | Pattern.isTop patt    = Pattern.top
   | Pattern.isBottom patt = Pattern.bottom
+  | not variableInTerm && not variableInPredicate = patt
+  | predicateIsBoolean =
+    TermLike.markSimplified (mkForall variable term)
```

Comment:
Should we check that the `term` is simplified here? Or do we only call this when we're sure it was simplified before.

Reply:
I ask @ttuegel the same question in another PR :)

I think we just assume that the input to all these functions (simplify, makeEvaluate, and so on) is already simplified.

Reply:
Ah, right. Found the conversation.